pkg/router/template/util/haproxytime/duration_parser_test.go

```diff
@@ -0,0 +1,94 @@
+package haproxytime_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/openshift/router/pkg/router/template/util/haproxytime"
+)
+
+func Test_ParseDuration(t *testing.T) {
+	tests := []struct {
+		input            string
+		expectedDuration time.Duration
+		expectedErr      error
+	}{
+		// Syntax error test cases.
+		{" 1m", 0, haproxytime.SyntaxError},
+		{"1m ", 0, haproxytime.SyntaxError},
+		{"1 m", 0, haproxytime.SyntaxError},
+		{"m", 0, haproxytime.SyntaxError},
+		{"", 0, haproxytime.SyntaxError},
+		{"+100", 0, haproxytime.SyntaxError},
+		{"-100", 0, haproxytime.SyntaxError},
+		{"-1us", 0, haproxytime.SyntaxError},
+		{"/", 0, haproxytime.SyntaxError},
+		{"123ns", 0, haproxytime.SyntaxError},
+		{"invalid", 0, haproxytime.SyntaxError},
+
+		// Validate default unit.
+		{"1", 1 * time.Millisecond, nil},
+
+		// Small values for each unit.
+		{"1us", 1 * time.Microsecond, nil},
```

Comment:
Was there a reason to drop the boundary case (i.e., 0) for each of these units?

Reply:
We don't allow zero values anymore, I changed the regex to match the haproxy template regex, per https://github.com/openshift/router/pull/483#discussion_r1355909022

Reply:
Ah, ok. Thanks.

Reply:
I'm missing something here.  If the function doesn't/shouldn't allow zero values anymore, doesn't that mean we should have test cases that verify that the function gives an error result for zero values?

Reply:
The haproxy parsing method tested zero values, but I removed those from ParseDuration that were returning success.  I just added one back into ParseDuration that returns failure, as it should now.